### PR TITLE
Hide the Search Input and Display when Touching Magnifying Glass #1230

### PR DESCRIPTION
**Pull request by UpworkBartkoski**
### Details
1. Replaced `ChatSwitcherView` with a touchable magnifying glass icon with conditional rendering by `isChatSwitcherActive`
2. Cleaned up the code so that the isLogoVisible and isClearButtonVisible props are removed from the ChatSwitcherSearchForm component

### Fixed Issues
#1230 

### Tests

1. Touch magnifying glass icon at the top of sidebar
2. It will display search form
3. Click on clear button to back to previous

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web
Screen Shot 2021-01-14 at 3 25 17 PM
Screen Shot 2021-01-14 at 3 26 23 PM

#### Mobile Web
Screen Shot 2021-01-14 at 3 27 20 PM
Screen Shot 2021-01-14 at 3 28 14 PM

#### Desktop
Screen Shot 2021-01-14 at 3 30 05 PM
Screen Shot 2021-01-14 at 3 30 35 PM

#### Android
Screen Shot 2021-01-14 at 3 21 51 PM
Screen Shot 2021-01-14 at 3 29 37 PM

### CODE REVIEW

**Comment by marcaaron** [comment]
@UpworkBartkoski can you merge `master` branch please? The screenshots do not have the latest changes. e.g. this is what my sidebar looks like at the moment:

Screen Shot 2021-01-14 at 11 49 47 AM

The magnifying glass can go right next to the avatar picture.

**Comment by shawnborton** [comment]
The icon size should be 20x20. The fill color of the icon should be `themeColors.icon`

**Comment by UpworkBartkoski** [comment]
> @UpworkBartkoski can you merge `master` branch please? The screenshots do not have the latest changes. e.g. this is what my sidebar looks like at the moment:
> 
> Screen Shot 2021-01-14 at 11 49 47 AM
> 
> The magnifying glass can go right next to the avatar picture.

Merged

File: src/pages/home/sidebar/SidebarLinks.js [hunk 1]
```diff
@@ -115,10 +117,20 @@ const SidebarLinks = (props) => {
...
-                    onLinkClick={props.onLinkClick}
-                    isChatSwitcherActive={props.isChatSwitcherActive}
-                />
+                {!props.isChatSwitcherActive ? (
+                    <TouchableOpacity
+                        style={[styles.flexRow, styles.sidebarHeaderTop]}
+                        onPress={() => ChatSwitcher.show()}
+                    >
+                        <MagnifyingGlassIcon width={30} height={30} />
+                    </TouchableOpacity>
```
Comment (marcaaron) [inline src/pages/home/sidebar/SidebarLinks.js:126]: This can be moved in between these two component here:

https://github.com/Expensify/Expensify.cash/blob/91fbf6cb1ce5451f1530182ce98ef2dc5e9af664/src/pages/home/sidebar/SidebarLinks.js#L142-L143

File: src/pages/home/sidebar/ChatSwitcherView.js [hunk 2]
```diff
@@ -103,6 +101,10 @@ class ChatSwitcherView extends React.Component {
             ChatSwitcher.show();
             this.textInput.focus();
         }, ['meta'], true);
+
+        if (this.props.isChatSwitcherActive) {
+            this.updateSearch(this.state.search);
+        }
```
Comment (marcaaron) [inline src/pages/home/sidebar/ChatSwitcherView.js:107]: Why this change?

File: src/styles/styles.js [hunk 3]
```diff
@@ -325,6 +325,7 @@ const styles = {
     sidebarHeaderTop: {
         paddingHorizontal: 20,
         paddingVertical: 12,
+        minHeight: 64,
```
Comment (marcaaron) [inline src/styles/styles.js:328]: This is probably not necessary after the icon is put into the new spot.

**Review by marcaaron — CHANGES_REQUESTED** [review]

File: src/pages/home/sidebar/SidebarLinks.js (hunk 1 above)
Comment (UpworkBartkoski) [reply src/pages/home/sidebar/SidebarLinks.js:126]: If it is moved, there will be an issue like below.
Screen Shot 2021-01-14 at 4 08 32 PM
Screen Shot 2021-01-14 at 4 11 09 PM

File: src/styles/styles.js (hunk 3 above)
Comment (UpworkBartkoski) [reply src/styles/styles.js:328]: removed `minHeight: 64`

File: src/pages/home/sidebar/SidebarLinks.js (hunk 1 above)
Comment (marcaaron) [reply src/pages/home/sidebar/SidebarLinks.js:126]: If you push your changes I will try to help resolve.

File: src/pages/home/sidebar/ChatSwitcherView.js (hunk 2 above)
Comment (UpworkBartkoski) [reply src/pages/home/sidebar/ChatSwitcherView.js:107]: `updateSearch` function sets state `options` as props of the `ChatSwitcherList`.
If options is empty, `ChatSwitcherList` is not rendered.
`updateSearch` was fired when focus on search input before.
Now, it should be called when `ChatSwitcherView` component mounted, because this component receive `isChatSwitcherActive: true` when clicking magnifying glass icon.

File: src/pages/home/sidebar/SidebarLinks.js (hunk 1 above)
Comment (UpworkBartkoski) [reply src/pages/home/sidebar/SidebarLinks.js]: pushed

File: src/pages/home/sidebar/ChatSwitcherView.js (hunk 2 above)
Comment (marcaaron) [reply src/pages/home/sidebar/ChatSwitcherView.js:107]: Alright. But why continue to check for `isChatSwitcherActive` it's _always_ active now isn't it? Looking at the code we won't render the `ChatSwitcherView` unless `isChatSwitcherActive` so it seems like a useless prop that can be removed.

File: src/pages/home/sidebar/SidebarLinks.js [hunk 4]
```diff
@@ -128,6 +124,22 @@ const SidebarLinks = (props) => {
             ]}
             >
                 <Header textSize="large" title="Chats" />
+                <View style={[chatSwitcherStyle]}>
```
Comment (marcaaron) [inline src/pages/home/sidebar/SidebarLinks.js:127]: The `View` with `chatSwitcherStyle` needs to go back where it was before. Only the `Touchable` + `MagnifyingGlassIcon` should be in between the `Header` and `AvatarWithIndicator`.

We don't need to explicitly hide the search icon in this case because it will be covered by the `ChatSwitcherView` once it renders. Apologies if this deviates from the original instructions slightly.

**Review by marcaaron — CHANGES_REQUESTED** [review]
Better thanks! But I still have some requests.

File: src/pages/home/sidebar/SidebarLinks.js (hunk 4 above)
Comment (UpworkBartkoski) [reply src/pages/home/sidebar/SidebarLinks.js]: Updated

**Review by marcaaron — CHANGES_REQUESTED** [review]

File: src/pages/home/sidebar/ChatSwitcherView.js [hunk 5]
```diff
@@ -103,6 +101,10 @@ class ChatSwitcherView extends React.Component {
             ChatSwitcher.show();
             this.textInput.focus();
         }, ['meta'], true);
+
+        if (this.props.isChatSwitcherActive) {
```
[the remark this answers is not in the document]
Comment (UpworkBartkoski) [reply src/pages/home/sidebar/ChatSwitcherView.js]: Correct!

File: src/pages/home/sidebar/SidebarLinks.js [hunk 6]
```diff
@@ -115,10 +117,12 @@ const SidebarLinks = (props) => {
...
         <View style={[styles.flex1, styles.h100, {marginTop: props.insets.top}]}>
             <View style={[chatSwitcherStyle]}>
-                <ChatSwitcherView
-                    onLinkClick={props.onLinkClick}
-                    isChatSwitcherActive={props.isChatSwitcherActive}
-                />
+                {props.isChatSwitcherActive && (
+                    <ChatSwitcherView
+                        onLinkClick={props.onLinkClick}
+                        isChatSwitcherActive={props.isChatSwitcherActive}
```
[the remark this answers is not in the document]
Comment (UpworkBartkoski) [reply src/pages/home/sidebar/SidebarLinks.js]: removed

File: src/pages/home/sidebar/SidebarLinks.js [hunk 7]
```diff
@@ -128,6 +132,14 @@ const SidebarLinks = (props) => {
             ]}
             >
                 <Header textSize="large" title="Chats" />
+                {!props.isChatSwitcherActive && (
```
[the remark this answers is not in the document]
Comment (UpworkBartkoski) [reply src/pages/home/sidebar/SidebarLinks.js]: Updated

File: src/pages/home/sidebar/ChatSwitcherView.js (hunk 2 above)
Comment (UpworkBartkoski) [reply src/pages/home/sidebar/ChatSwitcherView.js]: Correct!
Removed `isChatSwitcherActive`

**Comment by UpworkBartkoski** [comment]
> Better thanks! But I still have some requests.

Your requests done.

**Comment by UpworkBartkoski** [comment]
Updated

**Review by marcaaron — CHANGES_REQUESTED** [review]
Code changes look good, but there is an issue and the header is appearing below the sidebar now. I think if we conditionally render the header depending on `isChatSwitcherActive` then the problem will be resolved.

https://user-images.githubusercontent.com/32969087/104772131-06825080-5717-11eb-90de-685188dd6802.mp4

I also noticed a regression and the `KeyboardShortcut` in `ChatSwitcherView` no longer works. The best thing IMO would be to move it to the `HomePage.js`. This component also needs to focus after it mounts.

**Comment by marcaaron** [comment]
https://github.com/Expensify/Expensify.cash/blob/7823b07422f3bbc935f9e9f57f8e60e099c12a11/src/pages/home/sidebar/ChatSwitcherView.js#L101-L105

Can be moved and just call `this.textInput.focus();` instead in `ChatSwitcherView` `componentDidMount()`.

**Comment by UpworkBartkoski** [comment]
> Code changes look good, but there is an issue and the header is appearing below the sidebar now. I think if we conditionally render the header depending on `isChatSwitcherActive` then the problem will be resolved.
> 
>  2021-01-15_09-48-21.mp4 
> I also noticed a regression and the `KeyboardShortcut` in `ChatSwitcherView` no longer works. The best thing IMO would be to move it to the `HomePage.js`. This component also needs to focus after it mounts.

I didn't see this issue, may be I missed it.

**Comment by UpworkBartkoski** [comment]
> KeyboardShortcut

When the Command+K key being pressed in the status of `!props.isChatSwitcherActive`, then it displays search box.
But, focus can not be given to the search input.
Is this issue?
It should be focused on search input?

**Comment by UpworkBartkoski** [comment]
Updated.

**Comment by UpworkBartkoski** [comment]
E2E iOS Tests  `cocoapods > pod install` error.
How can I fix it?

**Review by marcaaron — APPROVED** [review]
LGTM thanks for those changes @UpworkBartkoski !
